### PR TITLE
Deploy to heroku

### PR DESCRIPTION
**Pull request by yashwp**

### CODE REVIEW

**Comment by mikevic** [comment]
@yashwp - could you also add in how to manage environment variables in Heroku?

File: domains/engineering/efficient/frontend_deployment.md [hunk 1]
```diff
@@ -242,3 +242,67 @@ d1sjyp0cxacal8.cloudfront.net`. This will take another 2-60 mins approximately.
...
+5. After deployment is successful, Heroku will generate a link to your application. For example *https://awesome--app.herokuapp.com*. This link will show **Application Error** for now.
+
+6. In case if you want to configure *Environment variables* for your application. Goto **Settings** tab > **Config Vars** add environment variable as key-value pair.
+
+    [![Environment variable](/assets/images/deploy-heroku-env.png)](/assets/images/deploy-heroku-env.png)
+
+### Configure your Angular application
+
+1. In your Angular application, open `package.json` file and copy `@angular/cli: <version>`,  `"@angular/compiler-cli": <version>` and `"typescript": <version>` from `devDependencies` to `dependencies` .
+
```
Comment (Manzurkds) [inline domains/engineering/efficient/frontend_deployment.md]: @yashwp Why do we need this step, it works even if they are in devDependencies if I'm not wrong?

File: domains/engineering/efficient/frontend_deployment.md [hunk 2]
```diff
@@ -242,3 +242,67 @@ d1sjyp0cxacal8.cloudfront.net`. This will take another 2-60 mins approximately.
...
+3. Now for Heroku to run your application we need to tell Heroku for node environment engines. Copy and paste the below code under `"devDependencies"` object.
+
+   ```json
+     "engines": {
+       "node": "<same version of your machine>",
+       "npm": "<same version of your machine>"
+     }
+   ```
+
```
Comment (Manzurkds) [inline domains/engineering/efficient/frontend_deployment.md]: @yashwp This should not be "under" `devDependencies` but "below" `devDependencies`, I hope it's a grammatical mistake😜

File: domains/engineering/efficient/frontend_deployment.md (hunk 1 above)
Comment (yashwp) [reply domains/engineering/efficient/frontend_deployment.md]: @Manzurkds This I'm not sure about. I haven't tried it removing. If it works without this step also, I remove it

**Pull request merged**